lib/src/commands/create.dart

```diff
@@ -142,9 +158,23 @@ Happy coding!
   }
 
   /// Install at_app_flutter to pub cache and set version constraints
-  Future<void> addDependency() async {
-    const retries = 2;
-    for (int i = 0; i < retries; i++) {
+  Future<void> cacheTemplatePackage({String? localPath}) async {
+    const tries = 2;
+    for (int i = 0; i < tries; i++) {
+      /// Get the package from the local path
+      if (localPath != null) {
+        ProcessResult result = await FlutterCli.pubAdd(
+          templatePackageName,
+          directory: projectDir,
+          localPath: localPath,
+        );
+
+        /// Exit code 65 is when the package is already included
+        if (result.exitCode == 65 || result.exitCode == 0) return;
+        continue;

```

Comment:
1) the retry is currently done immediately if the pub add fails. Maybe consider emitting a message and pausing for a second before doing the retry?
2) continue is not required as it is at the end of the for loop

Reply:
1. Colin had a one-off issue where the package failed to install, but was not reproducible. My assumption was that there was some sort of network issue, the idea behind the retry was just in case this happens again.

2. Good catch, by product from refactoring.

Reply:
Sure, I get why the retry is there. Just wondering if a short delay might make it more useful. Fine either way